testutil/testclient/testtx/context.go

```diff
@@ -277,3 +280,107 @@ func NewAnyTimesTxTxContext(
 
 	return txCtxMock, txCtx
 }
+
+// NewMockTxBuilder creates a new mock TxBuilder for testing purposes.
+// This mock can be used to verify transaction building operations.
+func NewMockTxBuilder(ctrl *gomock.Controller) *mockclient.MockTxBuilder {
+	txBuilder := mockclient.NewMockTxBuilder(ctrl)
+
+	// Create a var to store the fee for validation
+	var storedFee cosmostypes.Coins
+
+	// Setup necessary mock methods for tx building
+	txBuilder.EXPECT().SetMsgs(gomock.Any()).Return(nil).AnyTimes()
+
+	// When SetFeeAmount is called, store the fee
+	txBuilder.EXPECT().SetFeeAmount(gomock.Any()).DoAndReturn(func(fee cosmostypes.Coins) {
+		storedFee = fee
+	}).AnyTimes()
+
+	// When GetTx is called, return a mock with the stored fee
+	txBuilder.EXPECT().GetTx().DoAndReturn(func() cosmostypes.Tx {
+		return NewMockTx(storedFee)
+	}).AnyTimes()
+
+	txBuilder.EXPECT().SetTimeoutHeight(gomock.Any()).AnyTimes()
+	txBuilder.EXPECT().SetGasLimit(gomock.Any()).AnyTimes()
+
+	return txBuilder
+}
+
+// MockTx is a test implementation of the signing.Tx interface
```

Comment:
Did you consider using [`github.com/stretchr/testify/mock`](https://github.com/stretchr/testify?tab=readme-ov-file#mock-package)?